### PR TITLE
Feature/Upgrade flyway to 852

### LINKED ISSUE #5059: [SNYK: High] Remote Code Execution (RCE) (Due 03/18/2022)
**Overview**
[org.postgresql:postgresql](https://mvnrepository.com/artifact/org.postgresql/postgresql) is a Java JDBC 4.2 (JRE 8+) driver for PostgreSQL database. 

Affected versions of this package are vulnerable to Remote Code Execution (RCE) when using certain plugin features. pgjdbc instantiates plugin instances based on class names provided via authenticationPluginClassName, sslhostnameverifier, socketFactory, sslfactory, and sslpasswordcallback connection properties.

However, the driver did not verify if the class implements the expected interface before instantiating the class.

[SNYK-JAVA-ORGPOSTGRESQL-2390459](https://app.snyk.io/vuln/SNYK-JAVA-ORGPOSTGRESQL-2390459)

**Considerations:**
- This is tied to flyway. Perhaps we need to update flyway to 8.5.1. After upgrading check to see if postgresql driver is updated to 42.2.25 or 42.3.2 or higher.

**How to fix?**
- Upgrade org.postgresql:postgresql to version 42.2.25, 42.3.2 or higher.

### PR DESCRIPTION
**Pull request by fec-jli**
## Summary (required)

- Resolves #5059 

This PR will 
1)upgrade flyway to version 8.5.2 
2)be able to run snyk test on build.gradle.

### Required reviewers

1-2 developer, more welcome.

## How to test
PAET ONE: test flyway posgresslq driver (postgresql-42.2.25.jar) upgrade.
1. Test locally:
Ref: [https://github.com/fecgov/openFEC#installing-flyway](https://github.com/fecgov/openFEC#installing-flyway)
For MacOS user, open Safari browser go to [https://flywaydb.org/documentation/usage/commandline/](https://flywaydb.org/documentation/usage/commandline/)
and download `flyway-commandline-8.5.2-macosx-x64.tar.gz`
or you can download from google drive: [https://drive.google.com/drive/folders/1h5i4yL2pqlZc7ydSGahr_5ZZnVeUIGUJ](https://drive.google.com/drive/folders/1h5i4yL2pqlZc7ydSGahr_5ZZnVeUIGUJ)
update flyway conf file, activate virtue, run `invoke create_sample_db` no error.
2. Test on `dev` space, 
checkout feature branch, create your own test branch, deploy to dev space.
make sure without error and api works well.
Here is circleci log I deployed:
https://app.circleci.com/pipelines/github/fecgov/openFEC/1276/workflows/e22b1e6f-ed55-434d-a486-91f0d8487897/jobs/3777
snyk_circleci_log

3. Run snyk test locally to verify the vulnerable.
(1) you may need install Maven( `brew install maven`)
`mvn -v` #check Maven version
snyk_version
(2) Set Maven path in .bash.profile,
make sure you have: `export PATH="/usr/local/bin:$PATH"`

(3) Open new terminal, go to your local flyway-x.x.x/drivers/
run on flyway 8.4.3: `snyk test --scan-unmanaged --file=postgresql-42.2.19.jar`
(4) run on flyway 8.5.2: `snyk test --scan-unmanaged --file=postgresql-42.2.25.jar`
(5) Compare two results, you will see vulnerability gone.
snyk_843_error

snyk_852_noerror
 
PART TWO: run snyk test on build.gradle
1)If you checkout develop branch, go to data/flyway
run: `snyk test`
you will get error: "Could not get unknown property 'runtime' ..." because  gradle 7.4 ( `gradle -v`) change property name `runtime` to `runtimeonly`
see document [https://stackoverflow.com/questions/60340509/could-not-get-unknown-property-runtime-for-configuration-container-of-type-org](https://stackoverflow.com/questions/60340509/could-not-get-unknown-property-runtime-for-configuration-container-of-type-org)
snyk_gradle_error
2)check out this feature branch back
run: `snyk test` again.
you will be able to run it and get some vulnerabilities. We don't decide to fix these now.
snyk_gradle_issue

### CODE REVIEW

File: data/flyway/build.gradle [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
+	    exclude group: "mysql", module: "mysql-connector-java"
+        exclude group: "org.apache.derby", module: "derby"
+        exclude group: "org.apache.derby", module: "derbyclient"
+        exclude group: "org.hsqldb", module: "hsqldb"
+        exclude group: "org.mariadb.jdbc", module: "mariadb-java-client"
+        exclude group: "org.xerial", module: "sqlite-jdbc"
+        exclude group: "com.google.protobuf", module: "protobuf-java"
+        exclude group: "com.google.guava", module: "guava"
+        constraints {
+	       implementation("org.postgresql:postgresql:42.2.14.jre6")
```
Comment (pkfec) [inline data/flyway/build.gradle:24]: @fec-jli  postgresql:42.2.14.jre6 is not available in flyway-8.5.2. The postgresql jar  available in flyway-8.5.2 is postgresql-42.2.25.jar. Can you replace it?

Comment (pkfec) [reply data/flyway/build.gradle:24]: Here is the screenshot from [test/upgrade_flyway852 circle build](https://app.circleci.com/pipelines/github/fecgov/openFEC?branch=test%2Fupgrade_flyway852&filter=all)
Screen Shot 2022-03-08 at 9 22 11 PM

**Review by cnlucas — APPROVED** [review]
Works great for me! :)

**Review by pkfec — APPROVED** [review]
Good work @fec-jli

**Review by hcaofec — APPROVED** [review]
Great work!  Approved.

**Comment by fec-jli** [comment]
Check snyk [https://app.snyk.io/org/fecgov/project/e6c155e9-f0ac-4a49-98fa-83c24f5b74b3](https://app.snyk.io/org/fecgov/project/e6c155e9-f0ac-4a49-98fa-83c24f5b74b3)

**Pull request merged**